test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,623 @@
+        REQUIRE(bms->ext_cache->next[0] == ext);
+        validate_and_cleanup_ext_list(bms, 2);
+    }
+}
+
+TEST_CASE("Block session: __block_ext_discard", "[block_session]")
+{
+    WT_EXT *ext, *ext2, *ext3;
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+    REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext2) == 0);
+    REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext3) == 0);
+
+    // Construct extent cache with three items.
+    ext2->next[0] = ext3;
+    ext->next[0] = ext2;
+    bms->ext_cache = ext;
+    bms->ext_cache_cnt = 3;
+    SECTION("Discard every item in extent list")
+    {
+        REQUIRE(__ut_block_ext_discard(session->getWtSessionImpl(), 0) == 0);
+        validate_and_cleanup_ext_list(bms, 0);
+    }
+
+    SECTION("Discard until only one item is in extent list")
+    {
+        REQUIRE(__ut_block_ext_discard(session->getWtSessionImpl(), 1) == 0);
+
+        validate_and_cleanup_ext_list(bms, 1);
+    }
+
+    SECTION("Discard nothing in the extent list")
+    {
+        REQUIRE(__ut_block_ext_discard(session->getWtSessionImpl(), 3) == 0);
+        validate_and_cleanup_ext_list(bms, 3);
+    }
+
+    SECTION("Fake cache count and discard every item in extent list")
+    {
+        bms->ext_cache_cnt = 4;
+        REQUIRE(__ut_block_ext_discard(session->getWtSessionImpl(), 0) == WT_ERROR);
+    }
+}
+
+// Test unit size block functions.
+void
+validate_and_cleanup_size_block(WT_SIZE *size)
+{
+    REQUIRE(size != nullptr);
+    REQUIRE(size->depth == 0);
+    REQUIRE(size->off[0] == nullptr);
+    REQUIRE(size->size == 0);
+    REQUIRE(size->next[0] == nullptr);
+    __wt_free(nullptr, size);
+}
+
+void
+cleanup_size_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_SIZE *curr = bms->sz_cache;
+    WT_SIZE *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+void
+validate_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    REQUIRE(bms != nullptr);
+    WT_SIZE *curr = bms->sz_cache;
+
+    if (bms->sz_cache_cnt == 0)
+        REQUIRE(bms->sz_cache == nullptr);
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
+    for (int i = 0; i < expected_items; i++) {
+        REQUIRE(bms->sz_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+}
+
+void
+validate_and_cleanup_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    validate_size_list(bms, expected_items);
+    cleanup_size_list(bms);
+}
+
+TEST_CASE("Block session: __block_size_alloc", "[block_session]")
+{
+    WT_SIZE *sz = nullptr;
+
+    REQUIRE(__ut_block_size_alloc(nullptr, &sz) == 0);
+    validate_and_cleanup_size_block(sz);
+}
+
+TEST_CASE("Block session: __block_size_prealloc", "[block_session]")
```

Comment:
This file is too big to have an effective review, it may only be test code but already numerous issues have been called out. To have an effective discussion I think all of the functions being tested from here down should be moved to a subsequent ticket.

Reply:
I agree, i have split the ticket into multiple different tickets. This ticket will focus on the size functions only.